huxley/api/serializers/delegate.py

```diff
@@ -25,6 +31,32 @@ class Meta:
                   'session_three',
                   'session_four', )
 
+    def update(self, instance, validated_data):
+        if ('assignment' in validated_data and
+                validated_data['assignment'] != None and
+                not instance.assignment):
+
+            names = instance.name.split(' ')
+            username = names[0] + '_' + str(instance.id)
+            password = BaseUserManager().make_random_password(10)
+            user = User.objects.create(delegate=instance, username=username,\
+                                       user_type=User.TYPE_DELEGATE,\
+                                       last_login=datetime.now(),
+                                       first_name=names[0], last_name=names[-1],
+                                       email=instance.email)
+            user.set_password(password)
+            user.save()
+            send_mail('A new account has been created for {0}!\n'.format(instance.name),
```

Comment:
We should really use templates for emails. We have #109 open for that, if you wanna assign that to a new developer.

Reply:
I'll find someone this weekend to work on it